src/python/bot/startup/run_cron.py

```diff
@@ -55,12 +57,35 @@ def main():
   except ImportError:
     pass
 
+  # Few metrics get collected per job run, so
+  # no need for a thread to continuously push those
+  monitor.initialize(use_flusher_thread=False)
```

Comment:
Is there enough of a benefit to do this special casing ? 

If there's not much performance benefit, it's probably better to keep things more simple and uniform (i.e. always have a thread).

Reply:
It is mostly because of batch and swarming. Suppose a task takes 1m to run. We would have to wait for the entirety of FLUSH_INTERVAL_SECONDS (10m hardcoded) for them to dump, which would cause burn compute for no reason.

This thread is a weird concept for cron/batch/swarming, where only one task gets executed and the bot immediately quits. Hence this special casing, which IMO fits the ephemeral use cases better. The thread still makes sense for the use case of long lived bots, though.

Reply:
This is an interesting tradeoff, but 10 minutes seems tolerable for most of our cron jobs and 10 minutes is probably overly conservative anyway and can be reduced.

Reply:
For crons maybe, but for batch/swarming where there is stuff that runs under a minute, it is really hard for me to see value in using this thread.

Reply:
I tend to agree that as long as we make sure the thread flushes before stopping, it seems like a micro-optimization to avoid having a flusher thread.

Reply:
What value of FLUSH_INTERVAL_SECONDS should be chosen for this thread to be an universal solution?
If there is such a value that will prevent getting rate limited by the GCP metrics endpoint, and also not decrease the throughput of batch, I am fine with it.

However, this would take experimentation, which is not a really wise time investment. I would rather not have the trouble of choosing and stabilizing this value for short lived use cases.

Reply:
I think the interval isn't really critical to this decision. For shorter batch / cron jobs, could we instrument the interpreter exit instead to flush metrics? i.e. have `monitor.initialize()` register an atexit handler. 

That way, we keep things uniform in that we initialize this the same way, while not worrying about flush intervals.

Reply:
I can refactor this in the upcoming batch PR.
There is one last complexity: kubernetes can kill cronjobs due to timeout with a SIGTERM, followed by a SIGKILL. This SIGTERM has to be treated somehow

https://stackoverflow.com/questions/59973600/what-does-shutdown-look-like-for-a-kubernetes-cron-job-pod-when-its-being-termi